hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/prefetch/SingleFilePerBlockCache.java

```diff
@@ -268,12 +310,15 @@ public void close() throws IOException {
     int numFilesDeleted = 0;
 
     for (Entry entry : blocks.values()) {
+      entry.takeLock(Entry.LockType.WRITE);
```

Comment:
should we be acquiring locks in close()?
good: no race condition in close
bad) the usual

also, L303: should closed be atomic?

Reply:
> also, L303: should closed be atomic?

+1 to this suggestion, let me create a separate patch with HADOOP-18756 to better track it.

> good: no race condition in close
> bad) the usual

sounds reasonable, let me try setting timeout